### ISSUE TITLE
`style` prop doesn't support nested arrays

### ISSUE THREAD

**Issue by zuccha** [GeekyAnts/NativeBase#2419]
## I have gone through these following points
- [x]   Check latest documentation: https://docs.nativebase.io/
- [x]   Check for existing open/closed issues for a possible duplicate before creating a new issue: https://github.com/GeekyAnts/NativeBase/issues
- [x]   Use the latest NativeBase release: https://github.com/GeekyAnts/NativeBase/releases

## Issue Description

Native Base doesn't support nested arrays in `style` property. E.g.,
```
<View style={[[styles.foo]]} />
```

### node, npm, react-native, react and native-base version, expo version if used, xcode version
Node: 10.11.0
Yarn: 1.10.1
react: 16.5.0
react-native: 0.57.1
native-base: 2.8.1

### Expected behaviour

When passing styles in nested arrays, I should be able to see a red button with border width 2.

![correct](https://user-images.githubusercontent.com/19038023/47990178-29505d80-e0e7-11e8-8d9f-eaaf471cc994.png)

### Actual behaviour

Border and background styles are ignored.

![wrong](https://user-images.githubusercontent.com/19038023/47990186-31100200-e0e7-11e8-973a-1d0d4eba8527.png)

### Steps to reproduce
Code reproducible on expo.io.
```
import React from 'react';
import { Text, View, StyleSheet } from 'react-native';
import { Button } from 'native-base';
import { Constants } from 'expo';
import AssetExample from './components/AssetExample';
import { Card } from 'react-native-paper';

const Foo = () => (
  <View style={{flex: 1, alignSelf: 'center', justifyContent: 'center', alignItems: 'center'}}>
    <Button style={[{ justifyContent: 'center', width: 200 }, [styles.border, styles.bg]]}>
      <Text>Click here</Text>
    </Button>
  </View>
);

const styles = StyleSheet.create({
  border: { borderWidth: 2 },
  bg: { backgroundColor: 'red' }
});

export default class App extends React.Component {
  render() {
    return (
      <Foo />
    );
  }
}
```

### Is the bug present in both iOS and Android or in any one of them?

The bug is present both on iOS and Android.

### Workaround

In ES6, we can spread the array
```
<Button style={[{ justifyContent: 'center', width: 200 }, ...[styles.border, styles.bg]]}>
```

However, this is not the desired behaviour.

N.B.: We need to use nested arrays. Even if in the example above it is trivial to use styles directly without putting them in a nested array, in our application we compose styles programmatically, generating arrays of styles that we use in all our components.

**Comment by suvenduchhatoi** [comment]
@zuccha Closing the issue due to no response for a while. Hope your issue has been resolved.

**Comment by zuccha** [comment]
@suvenduchhatoi The extra brackets are the bug present in Native Base. Maybe I didn't explain well enough, but with more than one level of brackets styles don't work.

I tried to keep my example simple, so it is trivial to remove the extra brackets there, however I'm working on a project where we generate styles dynamically and it easier for us to have nested brackets (e.g., `style={[style1, [style2, style3]]}` instead of `style={[style1, style2, style3]}`).

As by default React Native supports nested brackets in styles, I would expect the same behaviour from Native Base.

I also tried by using `Text` from Native Base, but it doesn't solve the problem.
